node/src/components/storage.rs

```diff
@@ -340,6 +340,38 @@ impl Storage {
         }
     }
 
+    /// Reads from the state storage DB.
+    /// If key is non-empty, returns bytes from under the key. Otherwise returns `Ok(None)`.
+    /// May also fail with storage errors.
+    #[cfg(not(feature = "fast-sync"))]
+    pub(crate) fn read_state_store<K>(&self, key: &K) -> Result<Option<Vec<u8>>, Error>
+    where
+        K: AsRef<[u8]>,
+    {
+        let txn = self.env.begin_ro_txn()?;
+        let bytes = match txn.get(self.state_store_db, &key) {
+            Ok(slice) if slice.is_empty() => None,
```

Comment:
I think matching slices is supported now?
```suggestion
            Ok(&[]) => None,
```

Reply:
Done in [`6d6256a` (#904)](https://github.com/CasperLabs/casper-node/pull/904/commits/6d6256afbfb52e3c3bcc4ea817550a3ce25cea36).